### PR TITLE
Organization url & name updates

### PR DESCRIPTION
**Pull request by lomky**
Bumps #8169 

### Description
URL must exist and be unique.
Name must exist.

DB changes coming in another PR to finish ticket.

### CODE REVIEW

**Comment by lowellrex** [comment]
It looks like we've got a few failing tests caused by the new validations we added. I think we can resolve them by adding a `name` and `url` when we instantiate those objects (like [here, for instance](https://github.com/department-of-veterans-affairs/caseflow/blob/b61108e5c55fe3470f6d6834f56f6f642bba67e7/spec/controllers/organizations/tasks_controller_spec.rb#L11)).

**Comment by lomky** [comment]
Looks like I have a lot of tests to fix! VSO type are done, I'll dig into the rest tomorrow.

**Comment by lomky** [comment]
I believe I have gotten all the non-feature tests fixed. I also think I got all the feature tests as well, but I will let CircleCI confirm that for me once the FACOLs error is resolved

File: spec/models/appeal_spec.rb [hunk 1]
```diff
@@ -540,7 +540,12 @@
         let(:appeal) do
           create(:appeal, claimants: [create(:claimant, participant_id: participant_id_with_nil)])
         end
-        let!(:vso) { Vso.create(name: "Test VSO") }
+        let!(:vso) do
+          Vso.create(
```
Comment (pkarman) [inline spec/models/appeal_spec.rb]: There is a `vso` factory already in `spec/factories/organization.rb` -- maybe we make sure the default name and url are set there and use the factory instead?

File: spec/models/hearing_spec.rb [hunk 2]
```diff
@@ -76,7 +76,13 @@
     subject { hearing.assigned_to_vso?(user) }
 
     context "when the hearing is not assigned a vso" do
-      let(:vso) { Vso.create(participant_id: "999") }
+      let(:vso) do
+        Vso.create(
+          participant_id: "999",
```
Comment (pkarman) [inline spec/models/hearing_spec.rb]: same comment about factory.

File: spec/models/judge_team_spec.rb [hunk 3]
```diff
@@ -50,7 +50,11 @@
     # Limitation of the current approach is that users are effectively limited to being the admin of a single JudgeTeam.
     context "when user is admin of multiple JudgeTeams" do
       let!(:first_judge_team) { JudgeTeam.create_for_judge(user) }
-      let!(:second_judge_team) { JudgeTeam.create_for_judge(user) }
+      let!(:second_judge_team) do
```
Comment (pkarman) [inline spec/models/judge_team_spec.rb]: could we use a factory here? not sure if that would simplify.

**Pull request merged**